app/models/host/discovered.rb

```diff
@@ -54,13 +50,13 @@ def self.import_host facts
     # find existing or create new host record
     bootif_mac = FacterUtils::bootif_mac(facts).try(:downcase)
     hosts = ::Nic::Managed.where(:mac => bootif_mac, :primary => true)
-    if hosts.size == 0
+    if hosts.empty?
       host = Host.new(:name => hostname, :type => "Host::Discovered")
     else
       Rails.logger.warn "Multiple discovered hosts found with MAC address #{name_fact}, choosing one" if hosts.size > 1
       host = hosts.first.host
     end
-    host.type = "Host::Discovered"
+    raise ::Foreman::Exception.new(N_("Host already exists as managed: #{host.name}/#{bootif_mac}")) if host.type != "Host::Discovered"
```

Comment:
.new("Host already exists as managed: %s", "#{host.name}/#{bootif_mac}")

to generate the ERF code correctly?